src/native/credentials.c

```diff
@@ -32,6 +32,10 @@ struct aws_credentials *aws_credentials_new_from_java_credentials(JNIEnv *env, j
     jbyteArray session_token =
         (*env)->GetObjectField(env, java_credentials, credentials_properties.session_token_field_id);
 
+    if (access_key_id == NULL && secret_access_key == NULL) {
```

Comment:
Is there any other way to create credentials apart from using Credentials.java? If not, I think we just remove the or null check because partial nulls are not possible. 

Reply:
external projects might be using this directly. 
at the very least i think we should error out for defensive reasons to prevent bugs